source/material_model/damage_rheology.cc

```diff
@@ -0,0 +1,2028 @@
+      // limit viscosity variations due to temperature.
+      if (this->get_adiabatic_conditions().is_initialized())
+        {
+          const double adiabatic_energy_term
+            = exp((dislocation_activation_energy[phase_index] + dislocation_activation_volume[phase_index] * adiabatic_pressure)
+                  / (dislocation_creep_exponent[phase_index] * constants::gas_constant * this->get_adiabatic_conditions().temperature(position)));
+
+          const double temperature_dependence = energy_term / adiabatic_energy_term;
+          if (temperature_dependence > max_temperature_dependence_of_eta)
+            energy_term = adiabatic_energy_term * max_temperature_dependence_of_eta;
+          if (temperature_dependence < 1.0 / max_temperature_dependence_of_eta)
+            energy_term = adiabatic_energy_term / max_temperature_dependence_of_eta;
+        }
+
+      const double strain_rate_dependence = (1.0 - dislocation_creep_exponent[phase_index]) / dislocation_creep_exponent[phase_index];
+
+      return pow(dislocation_creep_prefactor[phase_index],-1.0/dislocation_creep_exponent[phase_index])
+             * std::pow(second_strain_rate_invariant,strain_rate_dependence)
+             * energy_term;
+    }
+
+
+
+    template <int dim>
+    double
+    DamageRheology<dim>::
+    viscosity (const double temperature,
+               const double pressure,
+               const std::vector<double> &composition,
+               const SymmetricTensor<2,dim> &strain_rate,
+               const Point<dim> &position) const
+    {
+      const SymmetricTensor<2,dim> shear_strain_rate = strain_rate - 1./dim * trace(strain_rate) * unit_symmetric_tensor<dim>();
+      const double second_strain_rate_invariant = std::sqrt(std::abs(second_invariant(shear_strain_rate)));
+
+      const double diff_viscosity = diffusion_viscosity(temperature, pressure, composition, strain_rate, position);
+
+      double effective_viscosity;
+      if (std::abs(second_strain_rate_invariant) > 1e-30)
+        {
+          const double disl_viscosity = dislocation_viscosity(temperature, pressure, composition, strain_rate, position);
+          effective_viscosity = disl_viscosity * diff_viscosity / (disl_viscosity + diff_viscosity);
+        }
+      else
+        effective_viscosity = diff_viscosity;
+
+      return effective_viscosity;
+    }
+
+
+
+    template <int dim>
+    double
+    DamageRheology<dim>::
+    enthalpy (const double      temperature,
+              const double      pressure,
+              const std::vector<double> &compositional_fields,
+              const Point<dim> &/*position*/) const
+    {
+      AssertThrow ((reference_compressibility != 0.0) || use_table_properties,
+                   ExcMessage("Currently only compressible models are supported."));
+
+      double enthalpy = 0.0;
+      if (n_material_data == 1)
+        enthalpy += material_lookup[0]->enthalpy(temperature,pressure);
```

Comment:
could as well be `enthalpy = ...`. same for the following functions.

(of course, only for the `if` branch -- in the `else` branch you have a loop)